### PR TITLE
CFE-3472: Updated documentation and examples to deprecate ifvarclass

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: examples/example-snippets/process_management.cf [hunk 1]
```diff
@@ -155,21 +156,21 @@ bundle agent process_restart
...
-      process_count   => up("cfservd");
+    "cfservd"
+      process_count => up("cfservd");
 
     cfservd_out_of_control::
       "cfservd"
-      signals         => { "stop" , "term" },
-      restart_class   => "start_cfserv";
+        signals       => { "stop" , "term" },
+        restart_class => "start_cfserv";
```
Comment (nickanderson) [inline examples/example-snippets/process_management.cf:173]: Can you alter this `start_cfserv` to `stopped_out_of_control_cfserv` while your here. It's bad practice IMHO to define classes that represent the /desired/ state. Instead, I think it's better to define classes that represent the current state. It makes policy read more easily:

e.g.

```
cfserved_out_of_control::
  signals => { "stop", "term" },
  restart_class => "stopped_cfserved_out_of_control";

commands:
  stopped_cfserved_out_of_control;
    "/var/cfenigne/bin/cf-serverd";

```

File: examples/example-snippets/restart_process_1.cf [hunk 2]
```diff
@@ -10,19 +10,21 @@ bundle agent process_restart
...
+        "cf-monitord",
+        "cf-serverd",
+        "cf-execd"
       };
+
   processes:
 
-      "$(component)"
+    "$(component)"
       restart_class => canonify("start_$(component)");
```
Comment (nickanderson) [inline examples/example-snippets/restart_process_1.cf:23]: here too, `s/start_/not_running/g`

File: reference/common-attributes-include.markdown [hunk 3]
```diff
@@ -14,7 +14,7 @@ the [Promise Types and Attributes] page. The common attributes are as follows:
 
 #### [handle][Promise Types and Attributes#handle]
 
-#### [ifvarclass][Promise Types and Attributes#ifvarclass]
```
Comment (nickanderson) [inline reference/common-attributes-include.markdown:17]: is ifvarclass no longer listed then"?

Comment (nickanderson) [reply reference/common-attributes-include.markdown:17]: i guess thats fine, if wasnt listed.

**Review by nickanderson — APPROVED** [review]
Overall good, Suggested a couple other cleanups that are a bit unrelated to whitespace or if/ifvarclass but not block worthy.

File: reference/common-attributes-include.markdown (hunk 3 above)
Comment (olehermanse) [reply reference/common-attributes-include.markdown:17]: Ye, it's just a list of links.

File: examples/example-snippets/process_management.cf (hunk 1 above)
Comment (olehermanse) [reply examples/example-snippets/process_management.cf:173]: Sure: https://github.com/cfengine/documentation/pull/2384

File: examples/example-snippets/restart_process_1.cf (hunk 2 above)
Comment (olehermanse) [reply examples/example-snippets/restart_process_1.cf:23]: Fixed: https://github.com/cfengine/documentation/pull/2384

**Pull request merged**